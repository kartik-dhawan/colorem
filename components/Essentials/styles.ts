export const styles = {
  essentialsSectionWrapper: {
    margin: {
      xs: "1.5rem 0rem",
      sm: "3rem 0rem",
    },
    backgroundColor: "#d9d6c9",
    color: "#3b4850",
    padding: {
      xs: "1rem 0rem",
      sm: "0.8rem 0rem",
      md: "0.4rem 0rem",
    },
  },
  essentialsSectionTitle: {
    textAlign: "center",
    color: "#111",
    padding: {
      xs: "1rem 1.5rem",
      lg: "1.5rem 1rem 1rem",
    },
    fontSize: {
      xs: "40px",
      md: "53px",
      lg: "60px",
    },
    fontWeight: 400,
    fontFamily: "Nunito Sans",
    letterSpacing: "0.4px",
  },
  essentialsSectionBodyItem: {
    display: "flex",
    padding: {
      xs: "1rem",
      md: "2rem 1.5rem",
      lg: "2rem",
    },
  },
  essentialsItemTitle: {
    fontWeight: {
      xs: 400,
      sm: 500,
    },
    padding: {
      xs: "5px 20px 5px 5px",
      sm: "5px 25px 5px 5px",
    },
  },
  essentialsItemBody: {
    display: "flex",
    alignItems: "center",
    fontSize: {
      xs: "18px",
      sm: "20px",
    },
  },
}
