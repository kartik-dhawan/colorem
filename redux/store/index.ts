import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
