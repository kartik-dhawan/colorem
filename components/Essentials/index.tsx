import { Grid, Typography } from "@mui/material"
import { Box } from "@mui/system"
import { Roboto_Condensed } from "next/font/google"
import { useSelector } from "react-redux"
import { RootType } from "../../redux/constants/stateTypes"
import { EssentialItemsType } from "../../utils/interfaces"
import { styles } from "./styles"

const roboto = Roboto_Condensed({
  subsets: ["latin"],
  weight: ["300", "400", "700"],
  style: ["normal", "italic"],
  display: "swap",
})

const Essentials = () => {
  const eid = "essentialsSection"
  const content = useSelector((state: RootType) => state.contentSlice.data)

  return (
    <Box
      className={eid + "Wrapper"}
      id={eid + "Wrapper"}
      data-testid={eid + "Wrapper"}
      sx={styles.essentialsSectionWrapper}
    >
      <Box
        className={eid + "Title " + roboto.className}
        id={eid + "Title"}
        sx={styles.essentialsSectionTitle}
      >
        The Essentials
      </Box>
      <Grid container className={eid + "Body"} id={eid + "Body"}>
        {content.essentialItemsText?.map((item: EssentialItemsType) => {
          return (
            <Grid
              item
              xs={12}
              md={6}
              lg={4}
              className={eid + "BodyItem"}
              id={eid + "BodyItem_" + item.id}
              sx={styles.essentialsSectionBodyItem}
              key={item.id}
            >
              <Typography variant="h2" sx={styles.essentialsItemTitle}>
                {item.id}
              </Typography>
              <Typography variant="body2" sx={styles.essentialsItemBody}>
                {item.body}
              </Typography>
            </Grid>
          )
        })}
      </Grid>
    </Box>
  )
}

export default Essentials
