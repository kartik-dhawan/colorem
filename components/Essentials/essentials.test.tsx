import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import store from "../../redux/store"
import { render } from "@testing-library/react"
import Essentials from "."

describe("The Essentials section Component", () => {
  it("-- should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Essentials />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("-- should render essentials section bullet points correctly", () => {
    const view = render(
      <Provider store={store}>
        <Essentials />
      </Provider>
    )
    const essentialsSectionBulletPoints = view.getByTestId(
      "essentialsSectionBody"
    )
    expect(essentialsSectionBulletPoints).toBeInTheDocument()
  })
})
