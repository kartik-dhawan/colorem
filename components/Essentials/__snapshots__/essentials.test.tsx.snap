// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Essentials section Component -- should match snapshot 1`] = `
<div
  className="essentialsSectionWrapper MuiBox-root css-1q5wgsi"
  data-testid="essentialsSectionWrapper"
  id="essentialsSectionWrapper"
>
  <div
    className="essentialsSectionTitle className MuiBox-root css-ns0x91"
    id="essentialsSectionTitle"
  >
    The Essentials
  </div>
  <div
    className="MuiGrid-root MuiGrid-container essentialsSectionBody css-11lq3yg-MuiGrid-root"
    data-testid="essentialsSectionBody"
    id="essentialsSectionBody"
  />
</div>
`;
